pkg/connector/roles.go

```diff
@@ -109,24 +108,23 @@ func (o *workspaceRoleType) List(
 		roleResource,
 	)
 	if err != nil {
-		return nil, "", nil, err
+		return nil, nil, err
 	}
-	return output, "", nil, nil
+	return output, &resources.SyncOpResults{}, nil
 }
 
 func (o *workspaceRoleType) Entitlements(
 	ctx context.Context,
 	resource *v2.Resource,
-	_ *pagination.Token,
+	attrs resources.SyncOpAttrs,
 ) (
 	[]*v2.Entitlement,
-	string,
-	annotations.Annotations,
+	*resources.SyncOpResults,
 	error,
 ) {
-	workspaceName, err := o.enterpriseClient.GetWorkspaceName(ctx, o.client, resource.ParentResourceId.Resource)
+	workspaceName, err := o.enterpriseClient.GetWorkspaceName(ctx, attrs.Session, o.client, resource.ParentResourceId.Resource)
```

Comment:
I think ideally we would get all the workspace names all at once whenever we don't have any, otherwise this will be very slow. It should be both a many put and a many get as it's otherwise a network request for each entry. Sorry if I hadn't explained that clearly earlier.